janus_core/calculations/phonons.py

```diff
@@ -0,0 +1,353 @@
+            chemical formula of the structure.
+        log_kwargs : Optional[dict[str, Any]]
+            Keyword arguments to pass to `config_logger`. Default is {}.
+        """
+        [minimize_kwargs, log_kwargs] = none_to_dict([minimize_kwargs, log_kwargs])
+
+        self.struct = struct
+        if struct_name:
+            self.struct_name = struct_name
+        else:
+            self.struct_name = self.struct.get_chemical_formula()
+
+        self.file_prefix = file_prefix if file_prefix else self.struct_name
+
+        # Ensure supercell is a valid list
+        self.supercell = [supercell] * 3 if isinstance(supercell, int) else supercell
+        if len(self.supercell) != 3:
+            raise ValueError("`supercell` must be an integer, or list of length 3")
+
+        self.displacement = displacement
+        self.t_step = t_step
+        self.t_min = t_min
+        self.t_max = t_max
+        self.minimize = minimize
+        self.minimize_kwargs = minimize_kwargs
+
+        self.log_kwargs = log_kwargs
+        self.log_kwargs.setdefault("name", __name__)
+        self.logger = config_logger(**self.log_kwargs)
+
+        if not self.struct.calc:
+            raise ValueError("Please attach a calculator to `struct`.")
+        self.calc = self.struct.calc
+        self.results = {}
+
+    def calc_phonons(self, write_results: bool = True) -> None:
+        """
+        Calculate phonons and optionally write results.
+
+        Parameters
+        ----------
+        write_results : bool
+            Whether to write out results to file. Default is True.
+        """
+        if self.minimize:
+            if self.logger:
+                self.minimize_kwargs["log_kwargs"] = {
+                    "filename": self.log_kwargs["filename"],
+                    "name": self.logger.name,
+                    "filemode": "a",
+                }
+            optimize(self.struct, **self.minimize_kwargs)
+
+        if self.logger:
+            self.logger.info("Beginning phonons calculation")
+
+        cell = self.ASE_to_PhonopyAtoms(self.struct)
+
+        supercell_matrix = (
+            (self.supercell[0], 0, 0),
+            (0, self.supercell[1], 0),
+            (0, 0, self.supercell[2]),
+        )
+        phonon = phonopy.Phonopy(cell, supercell_matrix)
+        phonon.generate_displacements(distance=self.displacement)
+        disp_supercells = phonon.supercells_with_displacements
+
+        phonon.forces = [
+            self.calc_forces(supercell)
+            for supercell in disp_supercells
+            if supercell is not None
+        ]
+
+        phonon.produce_force_constants()
+        phonon.run_mesh()
+        self.results["phonon"] = phonon
+
+        if self.logger:
+            self.logger.info("Phonons calculation complete")
+
+        if write_results:
+            self.results["phonon"].save(
+                f"{self.file_prefix}-params.yml", settings={"force_constants": True}
```

Comment:
will be usesful to write them in hdf5
somethins like phonon.save(filename=f"{sysname}-{model}.yaml", settings={"force_constants": False})
write_force_constants_to_hdf5(phonon.force_constants, filename=f"force_constants.hdf5")

from phonopy.file_IO import write_force_constants_to_hdf5

Reply:
Ok, why do we not want them in the same file?